examples/presigned-url-example.ts

```diff
@@ -61,6 +44,29 @@ const main = async () => {
   await runSignedTokenExample(signer, endpoint);
 };
 
+async function getSigningKey(): Promise<string> {
+  console.log('Checking for signing key in $MOMENTO_SIGNING_KEY');
+  if (process.env.MOMENTO_SIGNING_KEY) {
+    return process.env.MOMENTO_SIGNING_KEY;
+  }
+  // There is a limit of 5 signing keys per user, so it's best practice to
+  // create a signing key and store it somewhere secure. For convenience,
+  // this example creates a new signing key if one is not set in the
+  // environment variable `MOMENTO_SIGNING_KEY`. Signing keys can also be
+  // created and revoked via the Momento CLI.
+  console.log(
+    'Did not find signing key in environment, creating new signing key'
+  );
+  const signingKeyResponse = await momento.createSigningKey(60 * 24); // valid for 24 hours
+  if (signingKeyResponse instanceof CreateSigningKey.Error) {
+    throw signingKeyResponse.innerException();
+  }
+  process.env.MOMENTO_ENDPOINT = (
+    signingKeyResponse as CreateSigningKey.Success
```

Comment:
I'd really like to avoid casts in the example code, can we just put this in another branch of the `if` so that we get the smart cast?